vue/vite.config.ts

```diff
@@ -1,35 +1,41 @@
-import { defineConfig } from "vite";
+import { defineConfig, loadEnv } from "vite";
 import vue from "@vitejs/plugin-vue";
 import { execSync } from "child_process";
 
-export default defineConfig(() => {
-  // Load env file based on `mode` in the current working directory.
-  // Set the third parameter to '' to load all env regardless of the `VITE_` prefix.
-  const commitDate = execSync("git log -1 --format=%cI").toString().trimEnd();
-  const branchName = execSync("git rev-parse --abbrev-ref HEAD")
-    .toString()
-    .trimEnd();
-  const commitHash = execSync("git rev-parse HEAD").toString().trimEnd();
-  const lastCommitMessage = execSync("git show -s --format=%s")
-    .toString()
-    .trimEnd();
+export default ({ mode}) => {
+  // Loads all environment variables to see if we are in VITE_DOCKER_DEVELOPMENT
+  process.env = {...process.env, ...loadEnv(mode, process.cwd())};
+    // Load env file based on `mode` in the current working directory.
+    // Set the third parameter to '' to load all env regardless of the `VITE_` prefix.
+    const commitDate = execSync("git log -1 --format=%cI").toString().trimEnd();
+    const branchName = execSync("git rev-parse --abbrev-ref HEAD")
+      .toString()
+      .trimEnd();
+    const commitHash = execSync("git rev-parse HEAD").toString().trimEnd();
+    const lastCommitMessage = execSync("git show -s --format=%s")
+      .toString()
+      .trimEnd();
 
-  process.env.VITE_GIT_COMMIT_DATE = commitDate;
-  process.env.VITE_GIT_BRANCH_NAME = branchName;
-  process.env.VITE_GIT_COMMIT_HASH = commitHash;
-  process.env.VITE_GIT_LAST_COMMIT_MESSAGE = lastCommitMessage;
-  return {
-    plugins: [vue()],
-    server: {
-      host: "0.0.0.0",
-      port: 8080,
-      proxy: {
-        "/api": {
-          target: "http://localhost:8000",
-          xfwd: true,
+    process.env.VITE_GIT_COMMIT_DATE = commitDate;
+    process.env.VITE_GIT_BRANCH_NAME = branchName;
+    process.env.VITE_GIT_COMMIT_HASH = commitHash;
+    process.env.VITE_GIT_LAST_COMMIT_MESSAGE = lastCommitMessage;
+    const dockerDev = process.env.VITE_DOCKER_DEVELOPMENT;
```

Comment:
What do you think about making this env var be something like `VITE_DEV_PROXY_TARGET`, and make it default to `localhost:8000`, and override it to `django:8000` in the docker-compose client service? Then, we don't need to add any conditional logic to the vite.config.ts which seems cleaner to me.

Reply:
I think we still need conditionals (in the simplified form of `process.env.VITE_DEV_PROXY_TARGET || 'localhost'` ) unless we add a separate .env to the client directory for the default localhost when you decide to run `npm run dev` outside of the container.

Or I may be misinterpreting what you mean here?


I also want it to explicitly use 8080 for the dev port in the container and 3000 when running outside the container.  Mostly to prevent port conflicts if you happen to be changing node package dependencies live and don't want to relaunch the container for it to update.

Reply:
Ah, I forgot you were trying to get it to default to 3000 when outside the container. In that case, I think my approach isn't any simpler. So i think the way you had it before is fine, or we can just leave it as is now, I'm fine with either.